projectile.el

```diff
@@ -118,6 +118,22 @@ using the native indexing method."
   :group 'projectile
   :type 'boolean)
 
+(defcustom projectile-kill-buffers-filter 'kill-all
+  "Determine which buffers are killed by `projectile-kill-buffers'.
+
+When the kill-all option is selected, kills each buffer.
+
+When the keep-processes option is selected, keep buffers associated to a
+process.
+
+When the kill-only-files option is selected, kill only the buffer
+associated to a file."
+  :group 'projectile
+  :type '(radio
+          (const :tag "All project files" kill-all)
+          (const :tag "Preserve process buffer" keep-processes)
```

Comment:
I don't get what's the difference between the final two options - keep-processes and and kill-only-files are more or less the same as something's either a file buffer or some special buffer (although those won't necessary have process attached to them).

Reply:
You are probably right that the difference may not make much sense in terms of usage. I made the customization point extensible to allow for more fine-grained control, while removing the keep-processes choice.